OpenRA.Mods.Common/Traits/Player/ProvidesPrerequisite.cs

```diff
@@ -76,7 +86,9 @@ void Update(Player owner, string faction)
 				enabled = info.Factions.Contains(faction);
 
 			if (info.RequiresPrerequisites.Any() && enabled)
-				enabled = owner.PlayerActor.Trait<TechTree>().HasPrerequisites(info.RequiresPrerequisites);
+				enabled = techTree.HasPrerequisites(info.RequiresPrerequisites);
```

Comment:
Querying the state of the tech tree beforeupdating the tech tree doesnt look right at all. I expect this is the real cause of your bug. If you drop everything else and run this at the end of tick then your problem might go away.